src/inc/til/unicode.h

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+namespace til
+{
+    static constexpr bool is_surrogate(const wchar_t wch) noexcept
+    {
+        return (wch & 0xF800) == 0xD800;
+    }
+
+    static constexpr bool is_leading_surrogate(const wchar_t wch) noexcept
+    {
+        return (wch & 0xFC00) == 0xD800;
+    }
+
+    static constexpr bool is_trailing_surrogate(const wchar_t wch) noexcept
+    {
+        return (wch & 0xFC00) == 0xDC00;
+    }
+
+    constexpr std::wstring_view utf16_next(std::wstring_view wstr) noexcept
+    {
+        auto it = wstr.begin();
+        const auto end = wstr.end();
+        auto p = L"\uFFFD";
+        size_t l = 1;
```

Comment:
Oh yeah that's a good point. I called them `ptr` and `len` now, because I think those two are pretty common abbreviations when referring to ranges in C/C++.